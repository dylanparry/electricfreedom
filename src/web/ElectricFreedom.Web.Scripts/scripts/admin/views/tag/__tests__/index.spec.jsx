import React from 'react';
import { object } from 'prop-types';
import createContext from 'react-router-test-context';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import TagSwitch from '../index';
import TagAdd from '../add';
import TagEdit from '../edit';
import TagList from '../list';

const setup = (pathname = '/') =>
{
  const options = {
    context: createContext({
      location: { pathname },
    }),
    childContextTypes: {
      router: object,
    },
  };
  const wrapper = mount(<TagSwitch />, options);

  return {
    wrapper,
    snapshot: toJson(wrapper),
  };
};

describe('TagSwitch', () =>
{
  test('render', () =>
  {
    const { snapshot } = setup();

    expect(snapshot).toMatchSnapshot();
  });

  test('route "/tags/add" renders TagAdd', () =>
  {
    const { wrapper } = setup('/tags/add');

    expect(wrapper.find(TagAdd).exists()).toBe(true);
  });

  test('route "/tags/:tagId" renders TagEdit', () =>
  {
    for (let i = 0; i <= 5; i += 1)
    {
      const { wrapper } = setup(`/tags/${i}`);

      const tagEdit = wrapper.find(TagEdit);
      expect(tagEdit.exists()).toBe(true);
      expect(tagEdit.props().match.params.tagId).toBe(i.toString());
    }
  });

  test('route "/tags" renders TagList', () =>
  {
    const { wrapper } = setup('/tags');

    expect(wrapper.find(TagList).exists()).toBe(true);
  });
});
