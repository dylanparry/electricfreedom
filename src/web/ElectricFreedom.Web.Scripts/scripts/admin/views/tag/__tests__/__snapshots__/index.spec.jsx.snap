// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagSwitch render 1`] = `
<tagSwitch>
  <Switch />
</tagSwitch>
`;
